packages/flutter_tools/lib/src/flx.dart

```diff
@@ -75,26 +75,79 @@ Future<Null> build({
 
   DevFSContent kernelContent;
   if (!precompiledSnapshot && previewDart2) {
+    final File fingerprintFile = fs.file('$depfilePath.fingerprint');
+    final List<String> inputPaths = <String>[
+      mainPath,
+    ];
+    final List<String> outputPaths = <String>[
+      applicationKernelFilePath,
+    ];
+
+    bool needBuild = true;
+    final List<File> fingerprintFiles = <File>[fingerprintFile, fs.file(depfilePath)]
+      ..addAll(inputPaths.map(fs.file))
+      ..addAll(outputPaths.map(fs.file));
```

Comment:
Why are we adding these output files also to the set of files that need to be finger printed?

Reply:
That is what existing logic in build aot use case does. But as you suggested, I don't think it makes sense.